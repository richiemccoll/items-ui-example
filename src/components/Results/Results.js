import React from 'react';
import { Button } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import List from '../List/List';

const Results = ({ results, clearSearch }) => {
    return (
        <div>
            <Button onClick={() => clearSearch()} basic color="blue">Back</Button>
            {results && <List data={results} />}
        </div>
    )
}

export default Results;

Results.PropTypes = {
    results: PropTypes.array,
    clearSearch: PropTypes.isRequired
}