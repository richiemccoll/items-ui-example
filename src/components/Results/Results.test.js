import React from "react";
import Results from "./Results";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import sinon from "sinon";
import chai from "chai";

const resultProps = {
  data: [
    {
      title: "test",
      description: "test",
      image: ""
    },
    {
      title: "test2",
      description: "test2",
      image: ""
    }
  ],
  clearSearch() {
    return;
  }
};

describe("Results Component", () => {
  it("should render correctly based on props", () => {
    const component = renderer.create(
      <Results
        results={resultProps.data}
        clearSearch={resultProps.clearSearch}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call clearSearch", () => {
    const clearSearch = sinon.spy();
    const wrapper = mount(
      <Results results={resultProps.data} clearSearch={clearSearch} />
    );
    wrapper.find("button").simulate("click");
    chai.expect(clearSearch).to.have.property("callCount", 1);
  });
});
