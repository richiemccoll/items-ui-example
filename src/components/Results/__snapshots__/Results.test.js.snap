// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Component should render correctly based on props 1`] = `
<div>
  <button
    className="ui blue basic button"
    onClick={[Function]}
    tabIndex={undefined}
  >
    Back
  </button>
  <div
    className="sc-ifAKCX egvpF"
  >
    <div
      className="sc-bdVaJa iyRwOl"
    >
      <div
        className="sc-bwzfXH gOHCgz"
      >
        <img
          alt="test"
          className="sc-htpNat emEaka"
          src=""
        />
      </div>
      <div
        className="sc-bxivhb gutIee"
      >
        <h2
          className="ui header"
        >
          test
        </h2>
        <h3
          className="ui header"
        />
      </div>
    </div>
    <div
      className="sc-bdVaJa iyRwOl"
    >
      <div
        className="sc-bwzfXH gOHCgz"
      >
        <img
          alt="test2"
          className="sc-htpNat emEaka"
          src=""
        />
      </div>
      <div
        className="sc-bxivhb gutIee"
      >
        <h2
          className="ui header"
        >
          test2
        </h2>
        <h3
          className="ui header"
        />
      </div>
    </div>
  </div>
</div>
`;
